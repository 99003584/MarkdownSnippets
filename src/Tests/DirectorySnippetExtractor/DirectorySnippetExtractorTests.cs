using System.Collections.Concurrent;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using MarkdownSnippets;
using Xunit;
using Xunit.Abstractions;

public class DirectorySnippetExtractorTests :
    XunitApprovalBase
{
    [Fact]
    public void Case()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "DirectorySnippetExtractor/Case");
        var extractor = new DirectorySnippetExtractor();
        var snippets = extractor.ReadSnippets(directory);
        AssertCaseInsensitive(snippets.Lookup);
        ObjectApprover.Verify(snippets, Scrubber.Scrub);
    }

    static void AssertCaseInsensitive(IReadOnlyDictionary<string, IReadOnlyList<Snippet>> dictionary)
    {
        Assert.True(dictionary.ContainsKey("Snippet"));
        Assert.True(dictionary.ContainsKey("snippet"));
    }

    [Fact]
    public void Nested()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "DirectorySnippetExtractor/Nested");
        var extractor = new DirectorySnippetExtractor();
        var components = extractor.ReadSnippets(directory);
        ObjectApprover.Verify(components, Scrubber.Scrub);
    }

    [Fact]
    public void Simple()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "DirectorySnippetExtractor/Simple");
        var extractor = new DirectorySnippetExtractor();
        var components = extractor.ReadSnippets(directory);
        ObjectApprover.Verify(components, Scrubber.Scrub);
    }

    [Fact]
    public void VerifyLambdasAreCalled()
    {
        var directories = new ConcurrentBag<string>();
        var targetDirectory = Path.Combine(AssemblyLocation.CurrentDirectory,
            "DirectorySnippetExtractor/VerifyLambdasAreCalled");
        var extractor = new DirectorySnippetExtractor(
            directoryFilter: path =>
            {
                directories.Add(path);
                return true;
            }
        );
        extractor.ReadSnippets(targetDirectory);
        ObjectApprover.Verify(directories.OrderBy(file => file), Scrubber.Scrub);
    }

    public DirectorySnippetExtractorTests(ITestOutputHelper output) :
        base(output)
    {
    }
}