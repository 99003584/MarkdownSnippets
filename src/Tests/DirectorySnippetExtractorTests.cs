using System.Collections.Concurrent;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using MarkdownSnippets;
using VerifyXunit;
using Xunit;

[UsesVerify]
public class DirectorySnippetExtractorTests
{
    [Fact]
    public Task Case()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "DirectorySnippetExtractor/Case");
        DirectorySnippetExtractor extractor = new();
        var snippets = extractor.ReadSnippets(directory);
        AssertCaseInsensitive(snippets.Lookup);

        return Verifier.Verify(snippets);
    }

    static void AssertCaseInsensitive(IReadOnlyDictionary<string, IReadOnlyList<Snippet>> dictionary)
    {
        Assert.True(dictionary.ContainsKey("Snippet"));
        Assert.True(dictionary.ContainsKey("snippet"));
    }

    [Fact]
    public Task Nested()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "DirectorySnippetExtractor/Nested");
        DirectorySnippetExtractor extractor = new();
        var snippets = extractor.ReadSnippets(directory);
        return Verifier.Verify(snippets);
    }

    [Fact]
    public Task Simple()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "DirectorySnippetExtractor/Simple");
        DirectorySnippetExtractor extractor = new();
        var snippets = extractor.ReadSnippets(directory);
        return Verifier.Verify(snippets);
    }

    [Fact]
    public Task VerifyLambdasAreCalled()
    {
        ConcurrentBag<string> directories = new();
        var targetDirectory = Path.Combine(AssemblyLocation.CurrentDirectory,
            "DirectorySnippetExtractor/VerifyLambdasAreCalled");
        DirectorySnippetExtractor extractor = new(
            shouldIncludeDirectory: path =>
            {
                directories.Add(path);
                return true;
            });
        extractor.ReadSnippets(targetDirectory);
        return Verifier.Verify(directories.OrderBy(file => file));
    }
}