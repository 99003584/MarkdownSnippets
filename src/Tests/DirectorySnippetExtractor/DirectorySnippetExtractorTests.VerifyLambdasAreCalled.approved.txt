{
  Directories: [
    {
      Path: 'root\\DirectorySnippetExtractor\VerifyLambdasAreCalled\subpath'
    }
  ]
}